src/test/java/com/company/elixr/springbootapplicationwithdocker/FileOperationControllerTest.java

```diff
@@ -0,0 +1,141 @@
+package com.company.elixr.springbootapplicationwithdocker;
+
+import com.company.elixr.springbootapplicationwithdocker.constants.Constants;
+import com.company.elixr.springbootapplicationwithdocker.controller.FileOperationController;
+import com.company.elixr.springbootapplicationwithdocker.model.FileInfo;
+import com.company.elixr.springbootapplicationwithdocker.responses.SuccessResponse;
+import com.company.elixr.springbootapplicationwithdocker.responses.SuccessResponseForGetById;
+import com.company.elixr.springbootapplicationwithdocker.service.FileOperationService;
+import org.junit.Before;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.nio.charset.StandardCharsets;
+
+@RunWith(MockitoJUnitRunner.class)
+@WebMvcTest(FileOperationController.class)
+public class FileOperationControllerTest {
+
+    FileInfo fileInfo = new FileInfo();
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private FileOperationService fileOperationService;
+
+    @Before
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+        fileInfo.setFileName("testFile.txt");
+        fileInfo.setUserName("Nathaniel13");
+    }
+
+    @Test
+    public void test_UploadFile() throws Exception {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("file", "testFile.txt",
+                "text/plain",
+                "This is the file content".getBytes(StandardCharsets.UTF_8));
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart("/upload")
+                        .file(sampleFile)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .characterEncoding("UTF-8")
+                        .param("username", "Alex12"))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+    }
+
+    @Test
+    public void test_UploadFile_NoFileProvided() throws Exception {
```

Comment:
Since you are not providing both parameters, the name should not mention just 'File'
```suggestion
    public void test_UploadFile_NoParamsProvided() throws Exception {
```

Reply:
Made the change.